### PR TITLE
Fix Observer Alert list

### PR DESCRIPTION
**Pull request by rh12**
refs: [MBL-17893](https://instructure.atlassian.net/browse/MBL-17893)
affects: Parent
release note: Fixed alert list not refreshing after an alert is read or dismissed.

## What's changed
- Parent app alert list now updates dynamically after opening an alert or dismissing it
- Tab bar badge now also updates dynamically
- Dismiss is animated now

### Chores
- rearranged ObserverAlert related files and moved most of them to Parent app

## Test plan
- Verify the blue dot which marks an unread alert disappears after the alert was opened
- Verify that dismissing an alert removes it from the list of alerts
- Smoke test alert list and its refresh

**HINT:** An easy way to create alerts is to create announcements in Teacher app.

## Checklist
- [ ] Follow-up e2e test ticket created
- [x] Tested on phone
- [ ] Tested on tablet

[MBL-17893]: https://instructure.atlassian.net/browse/MBL-17893?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Parent Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<details><summary>Teacher Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Build failed, skipping coverage check</td>
    </tr>
  </tbody>
</table>



#### Release Note: 
Fixed alert list not refreshing after an alert is read or dismissed.

#### Affected Apps: Parent

[MBL-17893](https://instructure.atlassian.net/browse/MBL-17893)

<details><summary>:x: Swift lint</summary>


    ❌ /Users/vagrant/git/Parent/ParentUnitTests/ObserverAlerts/Model/UseCase/GetObserverAlertsTests.swift:29:1: Trailing Whitespace Violation: Lines should not have trailing whitespace (trailing_whitespace)
    
    
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ea89d22e2dbc53956e39ff3c430aa41ca9eceb7c
</p>

File: Parent/Parent/ObserverAlerts/ObserverAlertListViewController.swift [hunk 1]
```diff
@@ -161,14 +163,50 @@ extension ObserverAlertListViewController: UITableViewDataSource, UITableViewDel
...
+                receiveValue: { [weak self] in
+                    guard let self, let index = alerts.firstIndex(where: { $0.id == id }) else { return }
+
+                    alerts.remove(at: index)
+                    tableView.deleteRows(at: [.init(row: index, section: 0)], with: .automatic)
+                    updateTabBarBadgeCount()
+                }
+            )
+            .store(in: &subscriptions)
+    }
```
Comment (vargaat) [inline Parent/Parent/ObserverAlerts/ObserverAlertListViewController.swift:209]: For future reference I would add some comments on why we need the local state updates.

**Review by vargaat — DISMISSED** [review]
QA+1

**Review by vargaat — APPROVED** [review]

**Review by vargaat — APPROVED** [review]

**Review by ndrsszsz — APPROVED** [review]
QA +1

**Pull request merged**